File: backend/src/routes/students.js
// ============================================
// Öğrenci API Routes
// GET /api/students - Tüm öğrenciler
// GET /api/students/:id - Öğrenci detayı
// POST /api/students - Yeni öğrenci
// ============================================

import { Router } from 'express';
import { PrismaClient } from '@prisma/client';

const router = Router();
const prisma = new PrismaClient();

// Tüm öğrencileri getir (son deneme bilgisi ile)
router.get('/', async (req, res) => {
  try {
    const students = await prisma.student.findMany({
      include: {
        exams: {
          orderBy: { examDate: 'desc' },
          take: 1,
          select: { totalNet: true, examDate: true, examName: true }
        },
        _count: { select: { exams: true } }
      }
    });
    res.json(students);
  } catch (error) {
    console.error('Öğrenci listesi hatası:', error);
    res.status(500).json({ error: error.message });
  }
});

// Öğrenci detayı (tüm denemeleri ile)
router.get('/:id', async (req, res) => {
  try {
    const student = await prisma.student.findUnique({
      where: { id: parseInt(req.params.id) },
      include: {
        exams: {
          orderBy: { examDate: 'desc' },
          include: {
            results: {
              include: { subject: true }
            }
          }
        }
      }
    });

    if (!student) {
      return res.status(404).json({ error: 'Öğrenci bulunamadı' });
    }

    res.json(student);
  } catch (error) {
    console.error('Öğrenci detay hatası:', error);
    res.status(500).json({ error: error.message });
  }
});

// Yeni öğrenci ekle
router.post('/', async (req, res) => {
  try {
    const { name, examType, avatar } = req.body;

    if (!name || !examType) {
      return res.status(400).json({ error: 'İsim ve sınav türü zorunludur' });
    }

    const student = await prisma.student.create({
      data: {
        name,
        examType,
        avatar: avatar || '📚'
      }
    });

    res.status(201).json(student);
  } catch (error) {
    console.error('Öğrenci oluşturma hatası:', error);
    res.status(500).json({ error: error.message });
  }
});

export default router;

File: backend/app/routes/students.py
"""
Öğrenci API Route'ları
"""
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from app.database import get_db
from app.models import Student

router = APIRouter(prefix="/api/students", tags=["Öğrenciler"])


def student_to_dict(s):
    return {
        "id": s.id,
        "name": s.name,
        "examType": s.exam_type,
        "avatar": s.avatar,
        "createdAt": s.created_at.isoformat() if s.created_at else None,
    }


@router.get("")
def list_students(db: Session = Depends(get_db)):
    """Tüm öğrencileri listele"""
    students = db.query(Student).order_by(Student.id).all()
    return [student_to_dict(s) for s in students]


@router.get("/{student_id}")
def get_student(student_id: int, db: Session = Depends(get_db)):
    """Öğrenci detayını getir"""
    student = db.query(Student).filter(Student.id == student_id).first()
    if not student:
        raise HTTPException(status_code=404, detail="Öğrenci bulunamadı")
    return student_to_dict(student)
